### PR TITLE
Remove Data.Id and Type.Id and identity.ToId

### PR DESCRIPTION
**Pull request by alissapajer**

### CODE REVIEW

File: it/src/main/resources/tests/unsupportedTests.test [hunk 1]
```diff
@@ -1,15 +1,15 @@
 {
-    "name": "show tests that likely have a bug prior to the connector",
+    "name": "select first five tests with 12 backends ",
```
Comment (alissapajer) [inline it/src/main/resources/tests/unsupportedTests.test]: We actually used to have 12 backends.

**Comment by alissapajer** [comment]
@djspiewak I'm pushing a couple more comments to get rid of `OldPrecise` and `Data.Binary`. Hold off on merge until those are in so that we just have a single breaking release that encompasses all of this.

**Comment by alissapajer** [comment]
@djspiewak `Data.Binary` is a bigger can of worms, so I've pushed what I intend to push for this PR.

File: common/src/main/scala/quasar/common/data/Data.scala [hunk 2]
```diff
@@ -435,7 +429,7 @@ object Data {
           DateTimeConstants.nanosecond -> Int(value.getNano))),
         EJsonType(TypeTag.LocalDateTime))).right
 
-      case LocalTime(value)      => E.inj(ejson.Meta(
+      case LocalTime(value)=> E.inj(ejson.Meta(
```
Comment (djspiewak) [inline common/src/main/scala/quasar/common/data/Data.scala]: One too many spaces removed!

File: frontend/src/main/scala/quasar/frontend/data/DataCodec.scala [hunk 3]
```diff
@@ -159,23 +110,7 @@ object DataCodec {
...
-   * within array" test, which is impossible to satisfy without this trick.
-   *
-   * Note that only decoding is composed in this way.  Encoding is still singular
-   * in its backtracking.  This is mostly because I'm lazy and we relaly don't
-   * need recursive backtracking on encoding... YET.  If you need to add this
-   * (i.e. because of a situation analogous to the decoding test I referenced),
-   * then you can add it using a very similar trick.
-   */
-  def NewPrecise(recDecode: Json => DataEncodingError \/ Data) = new DataCodec {
+  val Precise = new DataCodec {
```
Comment (djspiewak) [inline frontend/src/main/scala/quasar/frontend/data/DataCodec.scala]: Awwwww… We lost my fancy fixed-point trick! 😛

File: common/src/main/scala/quasar/common/data/Data.scala (hunk 2 above)
Comment (alissapajer) [reply common/src/main/scala/quasar/common/data/Data.scala]: Sure, I can fix the spacing :)

**Comment by alissapajer** [comment]
@djspiewak ready.

**Pull request merged**